trollmoves/server.py

```diff
@@ -380,19 +382,34 @@ def create_file_notifier(attrs, publisher):
 
     tmask = (pyinotify.IN_CLOSE_WRITE |
              pyinotify.IN_MOVED_TO |
-             pyinotify.IN_CREATE)
+             pyinotify.IN_CREATE |
+             pyinotify.IN_DELETE)
 
     wm_ = pyinotify.WatchManager()
 
     pattern = globify(attrs["origin"])
     opath = os.path.dirname(pattern)
 
+    if 'origin_inotify_base_dir_skip_levels' in attrs:
+        pattern_list = pattern.split('/')
+        pattern_join = os.path.join(*pattern_list[:int(attrs['origin_inotify_base_dir_skip_levels'])])
+        opath = os.path.join("/", pattern_join)
+        LOGGER.debug("Using {} as base path for pyinotify add_watch.".format(opath))
+
     def fun(orig_pathname):
         """Publish what we have."""
-        if not fnmatch.fnmatch(orig_pathname, pattern):
-            return
+        if os.path.exists(orig_pathname):
+            if not fnmatch.fnmatch(orig_pathname, pattern):
+                return
+            elif (os.stat(orig_pathname).st_size == 0):
+                # Want to avoid files with size 0.
+                return
+            else:
+                LOGGER.debug('We have a match: %s', orig_pathname)
         else:
-            LOGGER.debug('We have a match: %s', orig_pathname)
+            # Some systems create temporary file which is then deleted shortly after.
+            # Want to skip these files
+            LOGGER.debug('orig_pathname: {} has disapeared. skipping this.'.format(orig_pathname))
```

Comment:
Is there enough "delay" for the temp file to have been removed? Also, wouldn't the temporary filename _not_ match the configured filename pattern?

Reply:
Unfortunately there is enough delay. I have a 3 party system that do that.

But you are right, it should _not_ match. I added this some years ago to my system so I dont remember completely why I added this. I need to check this.

Reply:
Here is what I get:
[2019-10-20 06:43:17,034 DEBUG    trollmoves.server] orig_pathname: /data/trollduction/METOP_level0/.IASI_HRP_00_M01_20191020063931Z_20191020064119Z_N_O_20191020063937Z.LXJt3P has disapeared. skipping this.

So if you match like this 
origin=/data/trollduction/METOP_level0/{sensor:s}_HRP_00_{platform_name:3s}_{start_time:%Y%m%d%H%M%S}Z_{end_time:%Y%m%d%H%M%S}Z_N_O_{process_time:%Y%m%d%H%M%S}Z

But anyway it would not match. Maybe I had a to greedy match string when I saw this and added this.

I think it it safe to remove it.